bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/CreateAdvBuilder.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *pa
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import java.util.concurrent.CompletableFuture;
+import org.apache.bookkeeper.client.AsyncCallback.CreateCallback;
+import org.apache.bookkeeper.client.BKException;
+
+/**
+ * Builder-style interface to create new ledgers
+ */
+public interface CreateAdvBuilder {
+
+    CreateAdvBuilder withLedgerId(int ledgerId);
+
+    WriteAdvHandler create() throws BKException, InterruptedException;
+
+    CompletableFuture<WriteAdvHandler> execute();
```

Comment:
execute should be an interface in itself, OpBuilder, which all of the ops implement.

Reply:
@ivankelly 
it makes sense. But I don't know if it will be really useful.
Client code will be like
bck.createLedgerOp().withXxx().execute()

the developer will never see explicitly the "Builder" interfaces and IMHO nobody will ever store temporary a generic Builder for later usage.
So I prefer not to introduce useless abstractions

Reply:
the interface is to provide consistency between all op builders. The user should be able to expect that there is always a execute call and it always returns a CompletableFuture. I wouldn't expect them to use the generic builder interface directly either.

Reply:
@ivankelly ok

Reply:
@ivankelly  I have introduced the OpBuilder interface but for open and delete it cannot be applied.
Do you think we should have this API for open/delete and have only an 'execute' method  ?

BK.openLedgerOp().withLedgerId(xxx).execute()

Reply:
we can pass the ledgerId to the first method as well
BK.openLedgerOp(ledgerId).execute()